MergerCli/Process.cs

```diff
@@ -118,16 +109,20 @@ private void ProcessBatch(IData baseData, List<Tile> newTiles, ref long tileProg
                 return;
             }
 
+            bool uploadOnly = baseData.IsNew || this._configManager.GetConfiguration<bool>("GENERAL", "uploadOnly");
+            bool shouldUpscale = !uploadOnly;
+
             for (int i = 0; i < newTiles.Count; i++)
             {
                 var newTile = newTiles[i];
                 var targetCoords = newTile.GetCoord();
                 List<CorrespondingTileBuilder> correspondingTileBuilders = new List<CorrespondingTileBuilder>()
                 {
-                    () => _getTileByCoord(targetCoords), () => newTile
+                    () => baseData.GetCorrespondingTile(targetCoords, shouldUpscale),
```

Comment:
why do you need to declare the "shouldUpscale" variable? why dont you just put the '!metadata.IsNewTarget' as this method arg?